src/Compiler/CompilerUtils.cpp

```diff
@@ -221,7 +223,8 @@ struct Command {
     llvm::sys::fs::make_absolute(cur_wdir, new_wdir);
     if (std::error_code ec = llvm::sys::fs::set_current_path(new_wdir)) {
       llvm::errs() << StringRef(new_wdir).str() << ": " << ec.message() << "\n";
-      exit(ec.value());
+      assert(ec.value() != 0 && "Expected nonnull error return value");
```

Comment:
Since we have an error, and its relying on `ec` being non null, but then return `ec.value()`, which must be nonzero to be then seen as an error, it seems ok to assert that it is nonzero. I will put a explanation.

Reply:
Looking at how `llvm::sys::fs::set_current_path` is implemented,
```
 std::error_code set_current_path(const Twine &path) {
   SmallString<128> path_storage;
   StringRef p = path.toNullTerminatedStringRef(path_storage);
  
   if (::chdir(p.begin()) == -1)
     return std::error_code(errno, std::generic_category());
  
   return std::error_code();
 }
```
you see that for both error or no error cases, it returns a std::error_code(). So the correct check should be,
```
std::error_code ec = llvm::sys::fs::set_current_path(new_wdir);
if (ec.value()) {
  llvm::errs() << StringRef(new_wdir).str() << ": " << ec.message() << "\n";
  return ec.value();
}
```